lib/puma/log_writer.rb

```diff
@@ -56,14 +57,17 @@ def self.null
 
     # Write +str+ to +@stdout+
     def log(str)
+      @stdout.is_a?(IO) and @stdout.wait_writable(1)
```

Comment:
https://ruby-doc.org/stdlib-2.7.1/libdoc/io/wait/rdoc/IO.html#wait_writable-method

>Waits until [IO](https://ruby-doc.org/stdlib-2.7.1/libdoc/io/wait/rdoc/IO.html) is writable without blocking and returns `self` or `nil` when times out.

Sounds like this shouldn't hurt to have

Reply:
> Sounds like this shouldn't hurt to have

My thought also.  Also, catching 'bad io' errors so a restart or shutdown isn't interrupted.

I intend to run it a few times in my fork to see if the intermittent error occurs...